pdns/mtasker_fcontext.cc

```diff
@@ -189,10 +189,12 @@ pdns_swapcontext
      or we switch back to pdns_swapcontext(),
      in both case we will be returning from a call to jump_fcontext(). */
 #if BOOST_VERSION < 106100
-    if (jump_fcontext (reinterpret_cast<fcontext_t*>(&octx.uc_mcontext),
-                       static_cast<fcontext_t>(ctx.uc_mcontext), 0)) {
-        std::rethrow_exception (ctx.exception);
-    }
+    intptr_t ptr = jump_fcontext(reinterpret_cast<fcontext_t*>(&octx.uc_mcontext),
+                                 static_cast<fcontext_t>(ctx.uc_mcontext), 0);
+
+    auto origctx = reinterpret_cast<pdns_ucontext_t*>(ptr);
+    if(origctx && origctx->exception)
+        std::rethrow_exception (origctx->exception);
```

Comment:
So, I understand why your fix works but I don't get why this didn't work in the first place. Shouldn't `origctx` be equal to `ctx` here, after `jump_fcontext()` returns, or am I getting this backward once again?

Reply:
so I tried that first, and printed the addresses of 'ctx', and they are different from the address of ctx in the MThread. I tried ctx and octx both. I admit this is somewhat of a "montecarlo programming" fix. There is a little bit of a dance around what 'ctx' means, and even the new boost code has to do some work to extract the ctx we returned from. Perhaps you don't know that when you do the jump_fcontext?

Reply:
Ah yes, I remember something like that, we can have different `pdns_ucontext_t` and of course the `fcontext_t` keeps changing.